### PR TITLE
MaterialXLoader: Implement separate(2|3|4) nodes

### LINKED ISSUE #29426: MaterialX: Add support for Separate2, Separate3, Separate4 nodes in MaterialXLoader
### Description

Separate nodes are not currently supported in the MaterialXLoader

### Solution

Support for separate2, separate3, and separate4 nodes

### Alternatives

I have not considered alternatives

### Additional context

I've created a [fork](https://github.com/beersandrew/three.js/tree/materialX-separate) with a [sample file](https://github.com/AcademySoftwareFoundation/MaterialX/commit/103c9212b7b8ba39fab0f0cfc7245c9ea657961b#diff-0d3ecede40f381c634d3e6e492723fa8f4d1c1dc989bad47960b442ecc663121)

If you run the fork locally and go to `/examples/?q=materialx#webgpu_loader_materialx` you can see this material shows up as black (2nd shaderball on the top row)  currently with an error that separate2 is not implemented. Images below of this state as well as the working state in the MaterialXGraphEditor

Error state
Screenshot 2024-09-16 at 16 47 04

Expected state
Screenshot 2024-09-16 at 16 55 06

### PR DESCRIPTION
**Pull request by aardgoose**
Fixed #29426 

Implement separate(2|3|4) nodes, this is the first multi output node as far as I know so possibly incorrect.
Appears to work with provided test sample.

@beersandrew can you test?

### CODE REVIEW

File: examples/webgpu_loader_materialx.html [hunk 1]
```diff
@@ -38,30 +38,31 @@
...
-				'standard_surface_copper.mtlx',
-				//'standard_surface_default.mtlx',
-				//'standard_surface_glass.mtlx',
-				//'standard_surface_glass_tinted.mtlx',
-				'standard_surface_gold.mtlx',
+//				'standard_surface_brass_tiled.mtlx',
+//				'standard_surface_brick_procedural.mtlx',
+				'separate2.mtlx',
+//				'standard_surface_carpaint.mtlx',
+/				//'standard_surface_chess_set.mtlx',
```
[the remark this answers is not in the document]
Comment (aardgoose) [reply examples/webgpu_loader_materialx.html]: fixed

[the remark this answers is not in the document]
Comment (aardgoose) [reply examples/webgpu_loader_materialx.html]: fixed

**Comment by beersandrew** [comment]
> Related issue: #29426
> 
> Implement separate(2|3|4) nodes, this is the first multi output node as far as I know so possibly incorrect. Appears to work with provided test sample.
> 
> @beersandrew can you test?

Hey, thanks for the quick response on this one, really appreciated!

I'm seeing some issues, I've added separate3.mtlx & separate4.mtlx samples in my MaterialX fork and I expect to see the following.

2,3,4 from left to right.

image

but I currently see this on the current branch:

image

not sure if this is helpful but it seems that when I step down into the nodes produced underneath the `colorNode` each `SplitNode` instance has `components` of `x`. 


Let me know if you're able to reproduce this with

```
const SAMPLE_PATH = 'https://raw.githubusercontent.com/beersandrew/MaterialX/separateNodeExample/resources/Materials/Examples/StandardSurface/';

const samples = [
  'separate2.mtlx',
  'separate3.mtlx',
  'separate4.mtlx'
];
```

**Comment by aardgoose** [comment]
@beersandrew 
I think I have fixed those cases, the initial input was being cached so that the outx was used repeatedly, and output elements weren't handled.  Both those fixed now.

**Comment by beersandrew** [comment]
@aardgoose looks good to me as well, thank you so much!

**Pull request merged**